### PR TITLE
Install and use clang in GPU test jobs.

### PR DESCRIPTION
**Pull request by ScottTodd**
Missed this when setting up these AMD jobs / migrating the NVIDIA job away from Docker. The jobs had been using gcc :P

ci-exactly: build_packages, test_amd_mi250, test_amd_mi300, test_amd_w7900, test_nvidia_t4

### CODE REVIEW

File: .github/workflows/pkgci_test_amd_mi250.yml [hunk 1]
```diff
@@ -49,6 +49,13 @@ jobs:
           ./build_tools/pkgci/setup_venv.py ${VENV_DIR} \
             --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
             --fetch-gh-workflow=${{ inputs.artifact_run_id }}
+      - name: "Installing build dependencies"
+        run: |
+          sudo apt update
+          sudo apt install cmake clang ninja-build
+          export CC=clang
+          export CXX=clang++
```
Comment (ScottTodd) [inline .github/workflows/pkgci_test_amd_mi250.yml]: Ohhhh wait, I should be using `>> $GITHUB_ENV`. Need to update a bunch of files o_o

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#environment-files

**Comment by ScottTodd** [comment]
Ping?

File: .github/workflows/pkgci_test_amd_mi250.yml [hunk 2]
```diff
@@ -30,11 +30,11 @@ jobs:
       IREE_HIP_DISABLE: 0
       IREE_HIP_TEST_TARGET_CHIP: "gfx90a"
```
Comment (saienduri) [inline .github/workflows/pkgci_test_amd_mi250.yml:31]: You can also add environment variables that you want to use here in the future

**Review by saienduri — APPROVED** [review]
LGTM thanks

**Pull request merged**